magento/magento2
# Fix inability to place PayPal guest order when Automatic Assignment to Customer Group is enabled

One of our shops reported that customers are unable to place a guest orders using the PayPal Express payment method. After some investigation it found out that this issue is caused by wiping out the customer email address from the quote model during the automatic assignment to the customer group. Moreover, the inability of placing the order is the case only when the PayPal is triggered from the cart by clicking "Checkout with PayPal" button. When one tries to make a guest order using PayPal in the payment method step in the checkout, the order is placed correctly due to the obvious reason, the e-mail address is explicitly set by the buyer during the 1st step of the checkout and thus it is correctly propagated to the corresponding customer object. This isn't the case when PayPal is started from the cart level.

### Description (*)
This PR implements setting an e-mail address on the customer object assigned to quote, which is prepared for the furthr processing by PayPal module when the checkout is recognized as a guest one. The e-mail address is set only when the corresponding customer object doesn't have it set yet.

### Fixed Issues (if relevant)
1. Fixes magento/magento2#25399 : Unable to place a guest order with PayPal Express when Enable Automatic Assignment to Customer Group is enabled

### Manual testing scenarios (*)
1. Configure and enable PayPal Express payment method, especially make sure that the button is shown in the cart
2. In the Magento admin configuration, go to Customers > Customer Configuration and set Enable Automatic Assignment to Customer Group to Yes
3. Make sure you are NOT logged in to your Magento customer account
4. Put any product to the cart and go to the cart page
5. Click "Checkout with PayPal" button, next login to your PayPal account and select payment instrument
6. Once you are redirected back to the order review page, select a shipping method and click "Place Order" button
7. The expected result is the order success page and the payment correctly processed by PayPal

### Questions or comments

### Contribution checklist (*)
